doc/code/qml_fourier.rst

```diff
@@ -0,0 +1,326 @@
+
+.. code::
+
+   fig, ax = plt.subplots(2, 1, sharey=True, figsize=(15, 4))
+   coefficients_violin_plot(coeffs, 2, ax, show_freqs=True);
+   plt.suptitle("Distribution of coefficients for circuit with weights", fontsize=16)
+      
+   
+.. image:: ../_static/fourier_vis_violin.png
+    :align: left
+    :width: 100%
+    :target: javascript:void(0);
+
+|
+	     
+A similar option is the :func:`~.pennylane.fourier.coefficients_box_plot`, which
+produces a plot of the same format but using a box plot.
+	    
+A different view can obtained using the	     
+:func:`~.pennylane.fourier.coefficients_radial_box_plot` function. This "rolls up"
+the coefficients onto a polar grid. Let us use it to visualize the same set of
+coefficients as above:
+
+.. code::
+
+   # The subplot axes must be *polar* for the radial plots
+   fig, ax = plt.subplots(
+       1, 2, sharex=True, sharey=True,
+       subplot_kw=dict(polar=True),
+       figsize=(15, 8)
+   )
+   coefficients_radial_box_plot(coeffs, 2, ax, show_freqs=True, show_fliers=False)
+   plt.suptitle("Distribution of coefficients for circuit with weights", fontsize=16)
+   plt.tight_layout()
+
+   
+.. image:: ../_static/fourier_vis_radial_box.png
+    :align: center
+    :width: 700px
+    :target: javascript:void(0);
+
+|
+
+The left plot displays the real part, and the right the imaginary
+part of the distribution over a parametrized quantum circuit's 
+Fourier coefficients. The labels on the "spokes" of the wheels represent the particular
+frequencies; we see that this matches the coefficients we found earlier. Note
+how the coefficient :math:`c_0` appears in the top middle of each plot; the
+negative frequencies extend counterclockwise from that point, and the positive
+frequencies increase in the clockwise direction. Such plots allow for a more
+compact representation of a large number of frequencies than the linear violin
+and box plots discussed above. For a large number of frequencies, however, it is
+recommended to disable the frequency labelling by setting ``show_freqs=False``,
+and hiding box plot fliers as was done above.
+
+Finally, for the special case of 1- or 2-dimensional functions, we can use the
+:func:`~.pennylane.fourier.coefficients_panel_plot` to plot the distributions of the
+sampled sets of Fourier coefficients on the complex plane.
+
+.. code::
+
+   # Need a grid large enough to hold all coefficients up to frequency 2
+   fig, ax = plt.subplots(5, 5, figsize=(12, 10), sharex=True, sharey=True)
+   coefficients_panel_plot(coeffs, 2, ax)
+   plt.suptitle(
+      "Fourier coefficients of circuit with weights in the complex plane",
+      fontsize=16
+   )
+   plt.tight_layout()
+
+.. image:: ../_static/fourier_vis_panel.png
+    :align: center
+    :width: 700px
+    :target: javascript:void(0);
+
+
+.. automodapi:: pennylane.fourier
+    :include-all-objects:
+    :no-inheritance-diagram:
```

Comment:
@glassnotes quick thought. Initially, our API pages were all just 'reference' lists of available functions, similar to https://numpy.org/devdocs/reference/routines.linalg.html, with minimal intros/descriptions.

However, the `qml.grouping`, `qml.kernels`, and `qml.fourier` are beginning to introduce more 'learning oriented' content into the API pages.

Question: should we move the list of functions to the _top_ of the page, so that users using this as a reference page can continue to quickly and easily find the function they need, with the overview placed _below_? The same way we place the examples at the bottom of the docstring, so that advanced users simply using the doc page as a reference lookup don't have to scroll.

Reply:
Actually, I just looked it up, and the [`np.fft`](https://numpy.org/doc/stable/reference/routines.fft.html) API page is a very good example of this: the functions are listed first for reference, and the 'background information' is below.

Reply:
Yes I'm on board with that! Even better though, we could port a large chunk of this content over to a how-to ("how to compute and visualize Fourier coefficients of quantum circuits"), or even a full demo. Then for this section we can leave only the very basic few lines explaining the purpose of the module.

Reply:
Oh I have no issue with the content this is great we can leave it here 😆 but definitely we could have a demo or how to in addition to this.